### PR TITLE
Add useful VS Code workspace settings

### PR DESCRIPTION
**Pull request by superkhau**
* VS Code: 80 char ruler

* VS Code: Tab completion

cc @bajtos @raymondfeng @ritch @superkhau

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
-    ".nyc_output": true
+    "packages/*/lib": true,
+    "packages/*/lib6": true
   },
-  "files.trimTrailingWhitespace": true,
   "files.insertFinalNewline": true,
-
-  "tslint.ignoreDefinitionFiles": true,
+  "files.trimTrailingWhitespace": true,
+  "tslint.autoFixOnSave": true,
```
Comment (raymondfeng) [inline .vscode/settings.json]: I won't enable this option by default as `tslint --fix` does not work reliably AFAIK.

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,25 +1,27 @@
 {
-  "typescript.tsdk": "./node_modules/typescript/lib",
+  "editor.insertSpaces": true,
```
Comment (raymondfeng) [inline .vscode/settings.json]: I think `true` is the default value anyway.

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
+  "editor.rulers": [80],
```
Comment (raymondfeng) [inline .vscode/settings.json]: 👍

**Comment by superkhau** [comment]
TY, I removed the auto fix on save rule and insertSpaces rule based on your feedback, landing after CI passes.

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
-  "files.trimTrailingWhitespace": true,
   "files.insertFinalNewline": true,
-
-  "tslint.ignoreDefinitionFiles": true,
+  "files.trimTrailingWhitespace": true,
   // Temporarily disable linting & compilation of examples,
   // because they are depending on features/APIs that are
   // not implemented yet
-  "tslint.exclude": "packages/examples/**/*"
+  "tslint.exclude": "packages/examples/**/*",
```
Comment (bajtos) [inline .vscode/settings.json]: FWIW, I think this line can be safely removed - there are no `packages/examples/**/*` files anymore.

**Pull request merged**